Organization APIs integration

jkhabra opened the pull request.

jkhabra committed:
fix: remove unneeded route

- Move the employee listing into components from pages

jkhabra committed:
fix: change basic layout into dashboard layout

jkhabra committed:
feat: create common client function

jkhabra committed:
feat: Implement organization create and listing APIs

- Create different component for listing

jkhabra committed:
feat: show error messages in add organization

jkhabra committed:
feat: show header in organization component

- Redirect on add organization from add button

bitspook commented on ui/src/components/Employees/List.tsx:
```diff
@@ -1,5 +1,5 @@
 import classNames from 'classnames/bind';
```
Shouldn't this page be `pages/employees/index.tsx`?

bitspook commented on ui/src/components/Employees/style.module.scss:
```diff
@@ -0,0 +1,101 @@
+  grid-template-rows: 26px 26px;
+}
+
+.employee-headshot-wrapper {
```
We are using both `container` and `wrapper` in class names. Both of them mean the same thing (i.e wrapper or container element). I suggest we choose one of these. Consistent names are extremely important.

bitspook commented on ui/src/components/Employees/style.module.scss:
```diff
@@ -0,0 +1,101 @@
+  margin-top: 10px;
+}
+
+.list-sction2-wrapper {
```
Type in the class name (`section2` is typed `sction2`)

bitspook commented on ui/src/components/Employees/style.module.scss:
```diff
@@ -0,0 +1,101 @@
+  padding: 15px;
+}
+
+.list-section1 {
```
Please justify your choice of name here. As an outsider, `list-section1` is revealing very little to me about what this class is about.

bitspook replied:
I am resolving all CSS related issues since we are going to change CSS anyway. I've created issues for implementing new styles.

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,30 @@
+const List: React.FC<FieldProps> = ({ ...p }) => {
+  const organizations = p.list.map((e) => (
+    <div className={c('list-content')} key={e.id}>
+      <div className={c('list-section1')}>
```
This class is named `list-section1`, but this is is the only section. What was the need to wrap the content of a list item in 2 `div`s?

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,30 @@
+  name: string;
+  banner?: string;
+  square_logo?: string;
+}
```
1. Name `Organizations` implies plural, i.e this is a collection, but it is not
2. We already have `Organization` type defined in `entities/Organization`. Please import that type instead of duplicating the type definition

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,30 @@
+  square_logo?: string;
+}
+
+interface FieldProps {
```
This name seem copied from some Field component. Please beware when copy-pasting code. Even an agnostic `Props` would work, but `FieldProps` is misleading.

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,30 @@
+}
+
+interface FieldProps {
+  list: Organizations[];
```
`list` is a very generic name. A variable's name should represent the intent behind its existence; ideally it should reflect what's stored in it, and what it might be used for. A better name here would be `organizations`.

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,30 @@
+    <div className={c('list-content')} key={e.id}>
+      <div className={c('list-section1')}>
+        <img className={c('item-wrapper')} src={e.square_logo} />
+        <span className={c('name-wrapper')}>{e.name}</span>
```
Both these names (`item-wrapper` and `name-wrapper`) seem misplaced. These elements aren't wrapping anything, but are the atomic elements themselves. I recommend we name them `item` and `name` instead. 

A name should reflect the intention behind the value named. A better name for `<img` class would perhaps be `logo`.

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,30 @@
+    </div>
+  ));
+
+  return <div className={c('list-container')}>{organizations}</div>;
```
CSS modules give us freedom to have non-colliding CSS. We don't need to namespace our classes anymore. i.e we don't need `list-` prefix for this component's classes.

bitspook commented on ui/src/containers/Layout/style.module.scss:
```diff
@@ -1,59 +1,106 @@
-  margin: 0 auto;
-  max-width: 1024px;
-  padding-top: 66px;
+  border-right: 1px solid $color-button-bg;
```
We are using `$color-button-bg` for coloring the border. Please check if there is an appropriate color for boundaries, and if not, please create a new color variable. Reusing variables in unrelated elements will likely end badly (e.g we might change the style of our buttons only to find of sidebar is not purple).

bitspook reviewed (changes requested):
I am ending the review after going through first 6 files. Please take a look at the comments, and do a self review to make sure everything is named more appropriately.

jkhabra committed:
fix: the class name issue

- change the wrapper into container and try to add meaningful name

jkhabra committed:
feat: add header component

jkhabra committed:
feat: implement employee create/fetch apis

jkhabra committed:
feat: implement designation fetch api

jkhabra committed:
feat: show employees and organization on sidebar for navigation

jkhabra committed:
fix: indentation in entities and rename fetch to fetchMany

- fix the naming in organization (Employee to Organization)

jkhabra committed:
remove commented code change the props variable in listing

jkhabra committed:
fix: style of add button

jkhabra committed:
fix: fetch APIs route

Commit:
Merge pull request #29 from sosinc/feature/employee-apis-integration

Employee apis integration

bitspook commented on ui/src/components/Employees/style.module.scss:
```diff
@@ -0,0 +1,101 @@
+  padding: 15px;
+}
+
+.section1 {
```
I recommend we avoid using super generic names, and name our classes based on semantics of their content.

bitspook commented on ui/src/components/Organization/List.tsx:
```diff
@@ -0,0 +1,25 @@
+}
+
+const List: React.FC<OrganizationProps> = ({ ...p }) => {
+  const organizations = p.organizations.map((e) => (
```
We can extract the function passed inside `map` here to outside this (i.e `List`) component, making it a component of its own (`ListItem`). It'll make code more readable, and also provide us some very small performance benefits (this function won't be recreated on every render).

bitspook commented on ui/src/containers/Layout/style.module.scss:
```diff
@@ -1,59 +1,136 @@
 @import 'src/scss/lib';
```
Let's rename this component to `DashboardLayout` instead.

bitspook commented on ui/src/containers/Layout/style.module.scss:
```diff
@@ -1,59 +1,136 @@
+  width: 20px;
+}
+
+.pic-container {
```
Let's extract Sidebar to its own component, and also rename `pic` to `avatar`. `avatar` more clearly conveys that this is related to user's avatar.

bitspook commented on ui/src/containers/Layout/style.module.scss:
```diff
@@ -1,59 +1,136 @@
+  cursor: pointer;
+  grid-column: 1;
+  grid-row: 3;
+}
```
Sidebar seem like an appropriate place to use flexbox since it requires a single-dimensional layout.

bitspook commented on ui/src/entities/User.ts:
```diff
@@ -1,6 +1,6 @@
-import { GraphQLClient } from 'graphql-request';
-import config from 'src/config';
+
+import client from 'src/lib/client';
```
I recommend we rename `src/lib/client` to `src/lib/graphqlClient`.

bitspook commented on ui/src/pages/dashboard/index.module.scss:
```diff
@@ -1,93 +1,30 @@
-  }
+  height: 1.5em;
+  margin-right: 2em;
+  width: 1.5em;
```
These look like they are leftover from styling Header. I suggest we remove them, and extract Header to its own component, and more appropriately name it  (e.g `TopBar`).

Commit:
chore: allow max-nesting-depth of 3 in SCSS

It is quite useful to nest styles for markup which is likely to be extracted out
as a separate component. max-nesting-depth of 1 is too rigid and pedantic anyways.

Commit:
fix: styles for no-organization page as per design

Commit:
feat: add NoItemsFound component

Commit:
refactor: use NoItemsFound component in employees/organizations listing

Commit:
refactor: minor refactoring here and there

Commit:
feat: add styles for buttons

Commit:
refactor: rename Layout to Dashboard layout

Commit:
refactor: use DashboardLayout in organization listing page

We are now keeping DashboardLayout header content in the consumer itself. It
will be extracted out to its own component once we have a better vision into how
it is getting used in different views.

jkhabra committed:
refactor: use dashboard layout in employees pages and add-organization

jkhabra committed:
fix: sidebar-layout styles

jkhabra committed:
wip: reset form on submit and disable the fields while submitting
